lib/logstash/inputs/imap.rb

```diff
@@ -81,10 +83,23 @@ def check_mail(queue)
       items.each do |item|
         next unless item.attr.has_key?("RFC822")
         mail = Mail.read_from_string(item.attr["RFC822"])
-        queue << parse_mail(mail)
+        if @strip_attachments
+          queue << parse_mail(mail.without_attachments!)
+        else
+          queue << parse_mail(mail)
+        end
       end
 
       imap.store(id_set, '+FLAGS', @delete ? :Deleted : :Seen)
+      
+      if @delete_expunge_on_process_msg 
+        # Force messages to be marked as "Deleted", the above may or may not be working as expected. "Seen" means nothing if you are going to
+        # delete a message after processing.
+        
+        imap.store(id_set, '+FLAGS', [:Deleted])
```

Comment:
We already have an option to set flags to :Deleted when `delete => true` is set. Any reason to also do this here?

Here's my proposal:

* Make this setting called `expunge`
* Call `imap.expunge` _after_ the `items.each` call so that we only have to call it once. Example: `imap.expunge if @expunge`